GetandTake/Services/Concrete/CategoryManager.cs

```diff
@@ -26,10 +27,31 @@ public async Task UpdateAsync(int categoryId, Category category)
             _repository.Update(category);
         }
     }
-    
-    public async Task<List<Category>> GetAllAsync() => 
+
+    public async Task<List<Category>> GetAllAsync() =>
         await _repository.GetItemsAsync();
 
-    public async Task<Category> GetByIdAsync(int categoryId) => 
+    public async Task<Category> GetByIdAsync(int categoryId) =>
         await _repository.GetAsync(category => category.CategoryID == categoryId);
-}
+
+    public async Task UploadImage(IFormFile file, int id)
+    {
+        var category = await GetByIdAsync(id);
+        if (category.ImagePath == null)
```

Comment:
I think you can just throw exceptions to avoid nesting, because it violates logic if you try to add new image to existing category. So, let's do it like:
```suggestion
       var category = await GetByIdAsync(id);
       
        if (category.ImagePath == null)
        {
              throw new Exception(...);
        }
        
        category.ImagePath = FileHelper.Add(file);
        _repository.Update(category);
```

(Same case for `EditImage`)

Reply:
I checked ImagePath on the Upload Page. Because the upload appears on this condition. So, I guess I don't need to check twice in here.